devito/propagator.py

```diff
@@ -202,19 +205,154 @@ def timings(self):
 
     @property
     def oi(self):
-        return self.profiler.oi
+        """Summary of operational intensities, by code section."""
 
-    @property
-    def oi_low(self):
-        return self.profiler.oi_low
+        gflops_per_section = self.gflops
+        bytes_per_section = self.traffic()
+        oi_per_section = {}
+
+        for i, subsection in enumerate(self.time_loop_stencils_b):
+            key = "%s%d" % (PRE_STENCILS.name, i)
+            oi_per_section[key] = 1.0*gflops_per_section[key]/bytes_per_section[key]
+
+        key = LOOP_BODY.name
+        oi_per_section[key] = 1.0*gflops_per_section[key]/bytes_per_section[key]
+
+        for i, subsection in enumerate(self.time_loop_stencils_a):
+            key = "%s%d" % (POST_STENCILS.name, i)
+            oi_per_section[key] = 1.0*gflops_per_section[key]/bytes_per_section[key]
+
+        return oi_per_section
 
     @property
-    def total_loads(self):
-        return self.profiler.num_loads
+    def niters(self):
+        """Summary of loop iterations, by code section."""
+
+        niters_per_section = {}
+
+        with_time_loop = lambda iters: self.nt*iters
+
+        for i, subsection in enumerate(self.time_loop_stencils_b):
+            key = "%s%d" % (PRE_STENCILS.name, i)
+            niters = subsection.limits[1] if isinstance(subsection, Iteration) else 1
+            niters_per_section[key] = with_time_loop(niters)
+
+        key = LOOP_BODY.name
+        niters = reduce(operator.mul, self.shape)
+        niters_per_section[key] = with_time_loop(niters)
+
+        for i, subsection in enumerate(self.time_loop_stencils_a):
+            key = "%s%d" % (POST_STENCILS.name, i)
+            niters = subsection.limits[1] if isinstance(subsection, Iteration) else 1
+            niters_per_section[key] = with_time_loop(niters)
+
+        return niters_per_section
+
+    def traffic(self, mode='realistic'):
+        """Summary of Bytes moved between CPU (last level cache) and DRAM,
+        by code section.
+
+        :param mode: Several estimates are possible: ::
+
+            * ideal: also known as "compulsory traffic", which is the minimum
+                number of bytes to be moved (ie, models an infinite cache)
+            * ideal_with_stores: like ideal, but a data item which is both read
+                and written is counted twice (load + store)
+            * realistic: assume that all datasets, even those that do not depend
+                on time, need to be re-loaded at each time iteration
+        """
```

Comment:
Can we make `ideal_with_stores` the default behaviour? `realistic` is more like the lower bound and I makes more sense (to me) to compare with the upper bound.